src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/ResetTests.cs

```diff
@@ -153,6 +153,31 @@ public async Task AppException_AfterHeaders_ResetInternalError()
                 .Build().RunAsync();
         }
 
+
+        [ConditionalFact]
+        [MinimumOSVersion(OperatingSystems.Windows, "10.0.19529", SkipReason = "Custom Reset support was added in Win10_20H2.")]
+        public async Task Goaway()
+        {
+            var deploymentParameters = GetHttpsDeploymentParameters();
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            await new HostBuilder()
+                .UseHttp2Cat(deploymentResult.ApplicationBaseUri, async h2Connection =>
+                {
+                    await h2Connection.InitializeConnectionAsync();
+
+                    h2Connection.Logger.LogInformation("Initialized http2 connection. Starting stream 1.");
+
+                    await h2Connection.StartStreamAsync(1, GetHeaders("/Goaway"), endStream: true);
+
+                    var frame = await h2Connection.ReceiveFrameAsync();
+                    Assert.Equal(Http2FrameType.GOAWAY, frame.Type);
```

Comment:
There should be a VerifyGoAway method that also checks the last stream id.
https://github.com/dotnet/aspnetcore/blob/df5269fea23f310fd7730b9eb0da5cc910cc1e6c/src/Servers/HttpSys/test/FunctionalTests/Http2Tests.cs#L166